lib/sitemap-stream.ts

```diff
@@ -1,79 +1,21 @@
-import { SitemapItem } from './sitemap-item';
 import { ISitemapItemOptionsLoose, ErrorLevel } from './types';
-import {
-  Transform,
-  TransformOptions,
-  TransformCallback,
-  Readable,
-  Writable,
-} from 'stream';
-import { ISitemapOptions, Sitemap } from './sitemap';
-export const preamble =
-  '<?xml version="1.0" encoding="UTF-8"?><urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:news="http://www.google.com/schemas/sitemap-news/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">';
-export const closetag = '</urlset>';
-export interface ISitemapStreamOpts
-  extends TransformOptions,
-    Pick<ISitemapOptions, 'hostname' | 'level' | 'lastmodDateOnly'> {}
-const defaultStreamOpts: ISitemapStreamOpts = {};
-export class SitemapStream extends Transform {
-  hostname?: string;
-  level: ErrorLevel;
-  hasHeadOutput: boolean;
-  lastmodDateOnly: boolean;
-  constructor(opts = defaultStreamOpts) {
-    opts.objectMode = true;
-    super(opts);
-    this.hasHeadOutput = false;
-    this.hostname = opts.hostname;
-    this.level = opts.level || ErrorLevel.WARN;
-    this.lastmodDateOnly = opts.lastmodDateOnly || false;
-  }
-
-  _transform(
-    item: ISitemapItemOptionsLoose,
-    encoding: string,
-    callback: TransformCallback
-  ): void {
-    if (!this.hasHeadOutput) {
-      this.hasHeadOutput = true;
-      this.push(preamble);
-    }
-    this.push(
-      SitemapItem.justItem(
-        Sitemap.normalizeURL(item, this.hostname, this.lastmodDateOnly),
-        this.level
-      )
-    );
-    callback();
-  }
-
-  _flush(cb: TransformCallback): void {
-    this.push(closetag);
-    cb();
-  }
+import { Transform, TransformOptions, TransformCallback, Readable } from 'stream';
+import { ISitemapOptions } from './sitemap';
+export declare const preamble = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><urlset xmlns=\"https://www.sitemaps.org/schemas/sitemap/0.9\" xmlns:news=\"https://www.google.com/schemas/sitemap-news/0.9\" xmlns:xhtml=\"https://www.w3.org/1999/xhtml\" xmlns:image=\"https://www.google.com/schemas/sitemap-image/1.1\" xmlns:video=\"https://www.google.com/schemas/sitemap-video/1.1\">";
```

Comment:
xmlns is not the same as a URL you would type into your browser. Changing this to https would produce an invalid document. xmlns must match the namespace declared in the xsd.

Reply:
Crud! I hate http so much, my insecurities get to me.  Just nuked them, submitted a blind pull request and here we are.  My apologies.